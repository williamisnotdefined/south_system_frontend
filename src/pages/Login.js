import React from 'react'

export default function Login() {
    return <h1>Oi Tesla :)~</h1>
}
