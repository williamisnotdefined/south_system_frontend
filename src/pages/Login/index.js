import React, { useContext } from 'react'

import { Creators as AuthActions, AuthContext } from '@containers/Auth'

import PageTitle from '@components/PageTitle'
import { Form, Submit } from '@components/Form'
import Input from '@components/Input'

import { Container } from './styles'

export default function Login() {
    const { state, dispatch, authenticateUser } = useContext(AuthContext)

    const onChangeUser = e => dispatch(AuthActions.setUser(e.target.value))

    const onChangePassword = e => dispatch(AuthActions.setPassword(e.target.value))

    const onSubmit = e => {
        e.preventDefault()
        authenticateUser()
    }

    return (
        <Container>
            <PageTitle>Login</PageTitle>
            <Form onSubmit={onSubmit}>
                <Input
                    value={state.user}
                    name="user"
                    onChange={onChangeUser}
                    placeholder="Type your user.."
                />

                <Input
                    type="password"
                    value={state.password}
                    name="password"
                    onChange={onChangePassword}
                    placeholder="Type your password.."
                />

                <Submit loading={state.loading}>Entrar</Submit>
            </Form>
        </Container>
    )
}
