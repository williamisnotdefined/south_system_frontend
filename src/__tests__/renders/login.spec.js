import React from 'react'
import renderer from 'react-test-renderer'

import { ThemeProvider } from 'styled-components'
import { theme } from '@styles'

import ContextContainer from '@containers'
import Login from '@pages/Login'

describe('Testing Login Page Render', () => {
    it('should render correctly', () => {
        const loginPage = renderer
            .create(
                <ThemeProvider theme={theme}>
                    <ContextContainer>
                        <Login />
                    </ContextContainer>
                </ThemeProvider>
            )
            .toJSON()

        expect(loginPage).toMatchSnapshot()
    })
})
