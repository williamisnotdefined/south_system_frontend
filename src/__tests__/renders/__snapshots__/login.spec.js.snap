// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login Page Render should render correctly 1`] = `
<div
  className="sc-AxheI bzmUgQ"
>
  <h1
    className="sc-AxjAm gwODik"
  >
    Login
  </h1>
  <form
    className="sc-AxhUy iwSbOU"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="sc-AxhCb hMENDW"
      name="user"
      onChange={[Function]}
      placeholder="Type your user.."
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="sc-AxhCb hMENDW"
      name="password"
      onChange={[Function]}
      placeholder="Type your password.."
      type="password"
      value=""
    />
    <button
      className="sc-AxgMl fgEWzF"
      type="submit"
    >
      Entrar
    </button>
  </form>
</div>
`;
